Formula/minimal-racket.rb

```diff
@@ -78,5 +85,16 @@ def caveats
       default-scope:
         installation
     EOS
+
+    # ensure Homebrew openssl is used
+    output = ""
+    on_macos do
+      output = shell_output("DYLD_PRINT_LIBRARIES=1 #{bin}/racket -e '(require openssl)' 2>&1")
+      assert_match(%r{loaded: .*openssl@1\.1/.*/libssl.*\.dylib}, output)
+    end
+    on_linux do
+      output = shell_output("LD_DEBUG=libs #{bin}/racket -e '(require openssl)' 2>&1")
+      assert_match(%r{init: .*openssl@1\.1/.*/libssl.*\.so}, output)
```

Comment:
Would using the `shared_library` DSL method simply things here?

I think the call you want is `shared_library("libssl", "*")` but this doesn't quite preserve some escape characters, since `shared_library` was built for globbing, not regex.

Reply:
> Would using the `shared_library` DSL method simply things here?

I don't think so. What I'm trying to match is:
```
macOS:  dyld: loaded: <1255353C-BF3B-3A26-93E4-C3E9510716C4> /usr/local/Cellar/openssl@1.1/1.1.1j/lib/libssl.1.1.dylib
Linux:  80795:	     calling init: /home/linuxbrew/.linuxbrew/opt/openssl@1.1/lib/libssl.so
```

For Linux, `shared_library` could let the match be rewritten as:
```
assert_match "init: #{Formula["openssl@1.1"].opt_lib}/#{shared_library("libssl")}"
```
which is certainly more precise than the current regex.

Unfortunately, I can't use the same trick for macOS, because [`Formula.prefix` is context-sensitive](https://rubydoc.brew.sh/Formula#prefix-instance_method), so `Formula.lib` in `test` blocks is exactly equal to `Formula.opt_lib`.